### PR TITLE
feat: no-invalid-properties rule

### PR DESCRIPTION
**Pull request by nzakas**
#### Prerequisites checklist

-   [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request?

Create a new `no-invalid-properties` rule.

#### What changes did you make? (Give an overview)

- Moved logic from `no-unknown-properties` into `no-invalid-properties`
- Moved logic from `no-invalid-property-values` into `no-invalid-properties`
- Added tests for `no-invalid-properties`
- Added docs for `no-invalid-properties`
- Removed old code for `no-unknown-properties` and `no-invalid-property-values`

#### Related Issues

Refs #9

### CODE REVIEW

**Comment by jeddy3** [comment]
Merging the rules looks good. It avoids that unnecessary check and means you can add syntax customisation at either the rule or language level.

You may want to add a valid test for `@property --foo { syntax: "*"; inherits: false; }`. I think the rule currently visits both [property declarations and descriptor declarations](https://drafts.csswg.org/css-syntax-3/#declaration) rather than just property ones and may incorrectly flag `syntax` and `inherits` as (unknown) properties.

**Comment by nzakas** [comment]
> I think the rule currently checks both [property declarations and descriptor declarations](https://drafts.csswg.org/css-syntax-3/#declaration) rather than just property ones.

Ah, nice catch! I didn't realize the same node type was used in both cases.

File: tests/rules/no-invalid-properties.test.js [hunk 1]
```diff
@@ -0,0 +1,181 @@
+/**
+ * @fileoverview Tests for no-invalid-property-values rule.
```
Comment (mdjermanovic) [inline tests/rules/no-invalid-properties.test.js:2]:
```suggestion
 * @fileoverview Tests for no-invalid-properties rule.
```

File: docs/rules/no-invalid-properties.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+}
+```
+
+## When Not to Use It
+
+If you aren't concerned with invalid properties, then you can safely disable this rule.
+
+## Prior Art
+
+\*[`declaration-property-value-no-unknown`](https://stylelint.io/user-guide/rules/declaration-property-value-no-unknown/)
```
Comment (mdjermanovic) [inline docs/rules/no-invalid-properties.md:51]:
```suggestion
-   [`declaration-property-value-no-unknown`](https://stylelint.io/user-guide/rules/declaration-property-value-no-unknown/)
```

**Review by mdjermanovic — APPROVED** [review]
LGTM, thanks!

**Pull request merged**